api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -98,39 +96,45 @@ public void validate(Object obj, Errors errors) {
 			        && !order.getDrug().getConcept().equals(order.getConcept())) {
 				errors.rejectValue("drug", "error.general");
 				errors.rejectValue("concept", "error.concept");
+				
 			}
 			if (order.getAction() != Order.Action.DISCONTINUE && order.getDosingType() != null) {
 				DosingInstructions dosingInstructions = order.getDosingInstructionsInstance();
 				dosingInstructions.validate(order, errors);
 			}
-			validateFieldsForOutpatientCareSettingType(order, errors);
-			validatePairedFields(order, errors);
-			validateUnitsAreAmongAllowedConcepts(errors, order);
-            validateForRequireDrug(errors, order);
-			ValidateUtil.validateFieldLengths(errors, obj.getClass(), "asNeededCondition", "brandName");
+			if (order.getDrug() == null && order.getDrugNonCoded() != null) {
```

Comment:
Hi ian@ibacher  made some changes on this here,could you be having some free few minutes during the course of the week and we try to get on call so as i demo to you my whole understanding of this ticket,?,i will be gratefull.regards

Reply:
Does this need to be up a little higher in this method? I.e., in the block from 91-93, if Drug is null, it will cause an error. I think this check needs to go before that. Something like:

```java
if (order.getDrug() == null) {
    if (order.getDrugNonCoded() != null) {
        order.setConcept(Context.getOrderService().getNonCodedDrugConcept());
    } else {
        ValidationUtils.rejectIfEmpty(errors, "concept", "error.null");
    }
} else if (order.getDrug().getConcept() == null) {
    ValidationUtils.rejectIfEmpty(errors, "concept", "error.null");
}
```